modules/core/src/lib/draw-layers.js

```diff
@@ -247,25 +252,60 @@ function drawLayersInViewport(
   logRenderStats({renderStats, pass, redrawReason, stats});
 }
 
+function getLightSources({viewport, effects}) {
```

Comment:
As mentioned above this logic needs to be inside the `LightEffect` instead of `draw-layers`. We will potentially have dozens of effects and this file should be indifferent to what they are.

Reply:
ditto